app/services/fact_cleaner.rb

```diff
@@ -0,0 +1,44 @@
+class FactCleaner
+  delegate :logger, :to => :Rails
+
+  def clean!
+    @to_delete = []
```

Comment:
the only public method is `clean!`, I might be missing something but every time someone calls it, it allocates new empty array so I don't see the potential issue.